src/server/game/Handlers/BankHandler.cpp

```diff
@@ -150,6 +150,152 @@ void WorldSession::HandleBuyBankSlotOpcode(WorldPackets::Bank::BuyBankSlot& pack
     _player->UpdateCriteria(CriteriaType::BankSlotsPurchased);
 }
 
+void WorldSession::HandleBuyReagentBankOpcode(WorldPackets::Bank::ReagentBank& reagentBank)
+{
+    if (!CanUseBank(reagentBank.Banker))
+    {
+        TC_LOG_DEBUG("network", "WORLD: HandleBuyReagentBankOpcode - %s not found or you can't interact with him.", reagentBank.Banker.ToString().c_str());
+        return;
+    }
+
+    if (_player->IsReagentBankUnlocked())
+    {
+        TC_LOG_DEBUG("network", "WORLD: HandleBuyReagentBankOpcode - Player (%s, name: %s) tried to unlock reagent bank a 2nd time.", _player->GetGUID().ToString().c_str(), _player->GetName().c_str());
+        return;
+    }
+
+    int64 price = 100 * GOLD;
+    _player->ModifyMoney(-price);
+    _player->UnlockReagentBank();
+}
+
+void WorldSession::HandleReagentBankDepositOpcode(WorldPackets::Bank::ReagentBank& reagentBank)
+{
+    if (!CanUseBank(reagentBank.Banker))
+    {
+        TC_LOG_DEBUG("network", "WORLD: HandleReagentBankDepositOpcode - %s not found or you can't interact with him.", reagentBank.Banker.ToString().c_str());
+        return;
+    }
+
+    if (!_player->IsReagentBankUnlocked())
+    {
+        _player->SendEquipError(EQUIP_ERR_REAGENT_BANK_LOCKED);
+        return;
+    }
+
+    // query all reagents from player's inventory
+    std::vector<Item*> items = _player->GetCraftingReagentItems();
+
+    if (!items.empty())
```

Comment:
you mustn't explicitly check if its empty if you are just iterating it. For loops are essentially while loops which check the condition before any action is done, and if the condition isnt satisfied the actions are not executed. That means if we have an empty vector the actions within the loop arent executed.

Reply:
ty 😄